README.md

```diff
@@ -49,21 +51,49 @@ Please, read the [getting started](https://doc.bblf.sh/using-babelfish/getting-s
 import bblfsh
 
 client = bblfsh.BblfshClient("0.0.0.0:9432")
-uast = client.parse("/path/to/file.py").uast
-print(uast)
-# "filter' allows you to use XPath queries to filter on result nodes:
-print(bblfsh.filter(uast, "//Import[@roleImport and @roleDeclaration]//alias"))
-
-# filter\_[bool|string|number] must be used when using XPath functions returning
-# these types:
-print(bblfsh.filter_bool(uast, "boolean(//*[@strtOffset or @endOffset])"))
-print(bblfsh.filter_string(uast, "name(//*[1])"))
-print(bblfsh.filter_number(uast, "count(//*)"))
+ctx = client.parse("/path/to/file.py")
+print(ctx)
+# or to get the results in a dictionary:
+resdict = ctx.get_all()
 
-# You can also iterate on several tree iteration orders:
-it = bblfsh.iterator(uast, bblfsh.TreeOrder.PRE_ORDER)
+# "filter' allows you to use XPath queries to filter on result nodes:
+it = ctx.filter("//python:Call")
 for node in it:
-    print(node.internal_type)
+    print(node)
+    # or:
+    doSomething(node.get())
+
+# filter must be used when using XPath functions returning these types:
+# XPath queries can return different types (dicts, int, float, bool or str), 
+# calling get() with an item will return the right type, but if you must ensure
+# that you are getting the expected type (to avoid errors in the queries) there
+# are alterative typed versions:
+x = next(ctx.filter("boolean(//*[@strtOffset or @endOffset])").get_bool()
+y = next(ctx.filter("name(//*[1])")).get_str()
+z = next(ctx.filter("count(//*)").get_int() # or get_float()
```

Comment:
This is not very Pythonic. IDK if it is hard to guess the type, but the following is more Pythonic:

```python
x = next(ctx.filter("boolean(//*[@strtOffset or @endOffset])").iterate())
for name in ctx.filter("name(//*[1])").iterate():
    print(name)
```

Where `iterate()` would be an unordered iterator over the resulting values. Or name it `results()`. Or remove it and iterate directly, anyway.

Reply:
Those are typed functions for queries returning boolean/string/integer/float values instead of nodes so the "typed get" is needed when the user doesn't have to do `isinstance` of the results to check that they're right every time.